src/java/org/apache/cassandra/cql3/statements/TruncateStatement.java

```diff
@@ -156,27 +156,41 @@ public TruncateStatement prepare(ClientState state, UnaryOperator<String> keyspa
             QualifiedName qual = qualifiedName;
             if (!ks.equals(qual.getKeyspace()))
                 qual = new QualifiedName(ks, qual.getName());
+            return driver.createTruncateStatement(rawCQLStatement, qual);
+        }
+    }
 
-            // local keyspaces are always local so don't use the remote statement even if it has been configured
-            if (!REMOTE_TRUNCATE_STATEMENT.isPresent() ||
-                SchemaManager.isKeyspaceWithLocalStrategy(super.keyspace()))
-            {
-                return new TruncateStatement(rawCQLStatement, qual);
-            }
-
-            Class<TruncateStatement> factoryClass = FBUtilities.classForName(REMOTE_TRUNCATE_STATEMENT.getString(),
-                                                                             "Remote truncate statement");
-            try
-            {
-                return factoryClass.getConstructor(String.class, QualifiedName.class)
-                                   .newInstance(rawCQLStatement, qual);
-            }
-            catch (NoSuchMethodException | IllegalAccessException | InstantiationException |
-                   InvocationTargetException e)
-            {
-                throw new RuntimeException("Unable to find correct constructor for " +
-                                           REMOTE_TRUNCATE_STATEMENT.getString(), e);
-            }
+    private static TruncateStatementDriver getDriverFromProperty()
+    {
+        try
+        {
+            return (TruncateStatementDriver)FBUtilities.classForName(TRUNCATE_STATEMENT_DRIVER.getString(),
+                                     "Truncate statement driver")
+                       .getConstructor().newInstance();
+        }
+        catch (NoSuchMethodException | IllegalAccessException | InstantiationException |
+               InvocationTargetException e)
+        {
+            throw new RuntimeException("Unable to find a truncate statement driver with name " +
+                                       TRUNCATE_STATEMENT_DRIVER.getString(), e);
+        }
+    }
+
+    private static final TruncateStatementDriver driver = TRUNCATE_STATEMENT_DRIVER.isPresent() ?
+                                                          getDriverFromProperty() :
+                                                          new DefaultTruncateStatementDriver();
+
+    public static interface TruncateStatementDriver
```

Comment:
just one nit - why is it called driver rather than provider ?

Reply:
Good catch! I was too focus on the implementation that I missed to care about the name :)

Reply:
Fixed